### ISSUE TITLE
Get test-dbadbcompression to have a table flag or a top N in table size

### ISSUE THREAD

**Issue by SQLDBAWithABeard** [sqlcollaborative/dbatools#3923]
### Summary of new feature

With databases with many many tables it can take a fair time for this beautiful command to run so @boosi83 and I think that it would be good to either specify a table or even better choose the top X in table size to be tested

### Latest version of dbatools as of writing

0.9.392

**Comment by sqllensman** [comment]
What would be a suitable name for the TopN parameter?

**Comment by MikeyBronowski** [comment]
Get-DbaTopResourceUsage uses -Limit however it has 'TOP' in the function name.

**Comment by sqllensman** [comment]
Get-DbaQueryExecutionTime uses the parameter MaxResultsPerDb. If the limit is to apply per database then I would think this would be more consistent.
Also when determining the Top N by table size should we
(a) Rank by TotalSpace
(b) Rank by UsedSpace
(c) Rank by Row count

Finally should this be done at
(1) Table level
(2) Index level 
(3) Partition level

**Comment by sqllensman** [comment]
@SQLDBAWithABeard 

I have created Pull request to add the requested functionality. Have added both options to filter by Schema, Table and no or objects.

While testing I noticed that 
(a) The procedure errors when used against a database with collation different to Server
(b) The code from the TigerToolbox was incorrect for Partitioned tables. The procedure sp_estimate_data_compression_savings returns values by Table, Index and Partition but join used to update results was ignoring partition

These have been fixed
